pgt/src/PGT.hs

```diff
@@ -66,21 +66,24 @@ parseShardConfig count@(ShardCount countValue) index@(ShardIndex indexValue) =
     then Left "Shard index ouside of shard count"
     else pure ShardConfig{..}
 
-selectShard :: ShardConfig -> Vector Test -> Vector Test
-selectShard ShardConfig{count=(ShardCount count), index=(ShardIndex index)} tests =
-    Vector.slice (convertImpure startIndex) (convertImpure effectiveChunkSize) tests
+selectShard :: ShardConfig -> Vector a -> Vector a
+selectShard ShardConfig{count=(ShardCount count), index=(ShardIndex index)} items =
+    Vector.slice (convertImpure startIndex) (convertImpure effectiveChunkSize) items
   where
     nominalChunkSize :: Natural
-    nominalChunkSize = convertImpure (Vector.length tests) `div` count
+    nominalChunkSize = length `div` count
```

Comment:
__question:__ I don't suppose it's possible for `count` to be `0`?

Reply:
yes, see the constructor of `ShardCount`.

Reply:
well `parseShardCount` you'll not find the constructor exported.

Reply:
@dkubb `parseShardCount` forbids that.